fetch.bs

```diff
@@ -1037,6 +1024,124 @@ worked on in <a href=https://github.com/whatwg/fetch/issues/1156>issue #1156</a>
  <var>body</var>.
 </ol>
 
+<hr>
+
+<p>To <dfn export for=body>incrementally read</dfn> a <a for=/>body</a> <var>body</var>, given an
+algorithm <var>processBodyChunk</var>, an algorithm <var>processEndOfBody</var>, an algorithm
+<var>processBodyError</var>, and an optional null, <a for=/>parallel queue</a>, or
+<a for=/>global object</a> <var>taskDestination</var> (default null), run these steps.
+<var>processBodyChunk</var> must be an algorithm accepting a <a for=/>byte sequence</a>.
+<var>processEndOfBody</var> must be an algorithm accepting no arguments. <var>processBodyError</var>
+must be an algorithm accepting an exception.
+
+<ol>
+ <li><p>If <var>taskDestination</var> is null, then set <var>taskDestination</var> to the result of
+ <a>starting a new parallel queue</a>.
+
+ <li>
+  <p>Let <var>reader</var> be the result of <a for=ReadableStream>getting a reader</a> for
+  <var>body</var>'s <a for=body>stream</a>.
+
+  <p class=note>This operation ought not to throw an exception.
+
+ <li><p>Perform the <a>incrementally-read loop</a> given <var>reader</var>,
+ <var>taskDestination</var>, <var>processBodyChunk</var>, <var>processEndOfBody</var>, and
+ <var>processBodyError</var>.
+</ol>
+
+<p>To perform the <dfn>incrementally-read loop</dfn>, given a {{ReadableStreamDefaultReader}} object
+<var>reader</var>, <a for=/>parallel queue</a> or <a for=/>global object</a>
+<var>taskDestination</var>, algorithm <var>processBodyChunk</var>, algorithm
+<var>processEndOfBody</var>, and algorithm <var>processBodyError</var>:
+
+<ol>
+ <li>
+  <p>Let <var>readRequest</var> be the following <a>read request</a>:
+
+  <dl>
+   <dt><a for="read request">chunk steps</a>, given <var>chunk</var>
+   <dd>
+    <ol>
+     <li><p>Let <var>continueAlgorithm</var> be null.
+
+     <li><p>If <var>chunk</var> is not a {{Uint8Array}} object, then set
+     <var>continueAlgorithm</var> to this step: run <var>processBodyError</var> given a
+     {{TypeError}}.
+
+     <li>
+      <p>Otherwise:
+
+      <ol>
+       <li><p>Let <var>bytes</var> be a <a lt="get a copy of the buffer source">copy of</a>
+       <var>chunk</var>.
```

Comment:
Maybe note that this copy can often be elided if the user agent is working on its own streams, or e.g. a stream supplied from a service worker where the copy has already been made while crossing the realm boundary? I dunno, maybe not worth it...